### PR TITLE
Shower now regen stamina.... Not for the felinids.

### PR DESCRIPTION
**Pull request by Xackii**
## About The Pull Request

Washing now give you status effect that regen 4 stamina per tick.

https://github.com/user-attachments/assets/1691ac4b-d8e4-402a-98d1-3cba61c00879

BUT if you felinid you will loss 4 stamina insteed becouse cats don't love water.

https://github.com/user-attachments/assets/e566e4d8-7f8a-47e6-aadc-b2910758d6ea

## Why It's Good For The Game

Gives more reasons to wash.

## Changelog
:cl:
add: Showers now heals stamina when you washing. But not for you catgitls.
/:cl:

### CODE REVIEW

File: code/game/objects/structures/shower.dm
Comment (MrMelbert) [inline code/game/objects/structures/shower.dm:254]:
```suggestion
	take_his_status_effect.remove_status_effect(/datum/status_effect/shower_regen)
```
`remove_status_effect` does nothing if the mob doesn't have the supplied effect

**Comment by Sadboysuss** [comment]
Does this mean ifI'm under a shower I can tank another baton hit? Also species change

File: code/game/objects/structures/shower.dm
Comment (MrMelbert) [inline code/game/objects/structures/shower.dm:266]:
```suggestion
	living_target.apply_status_effect(/datum/status_effect/shower_regen)
```
`apply_status_effect` will do nothing if they already have it because your status effect has `STATUS_EFFECT_UNIQUE`

File: code/datums/status_effects/neutral.dm
Comment (MrMelbert) [inline code/datums/status_effects/neutral.dm:615]:
```suggestion
	desc = "A good wash fills me with energy!"
```

File: code/datums/status_effects/neutral.dm
Comment (MrMelbert) [inline code/datums/status_effects/neutral.dm:620]:
```suggestion
	desc = "Waaater... Fuck this WATER!!"
```

File: code/game/objects/structures/shower.dm
Comment (Xackii) [reply code/game/objects/structures/shower.dm:266]: they thought of everything for me

**Comment by SmArtKar** [comment]
Are you sure this won't cause felinids to get stunlocked under showers

**Comment by Xackii** [comment]
> Are you sure this won't cause felinids to get stunlocked under showers

I don’t see any problems (Melbert made it so that it was impossible to take it into the infinity insta stun, I hope he thought of everything @MrMelbert )

**Comment by MrMelbert** [comment]
The diminishing returns should prevent this from perma stunning yes, should test it anyways though

File: code/datums/status_effects/neutral.dm
Comment (Sadboysuss) [inline code/datums/status_effects/neutral.dm:618]:
```suggestion
/atom/movable/screen/alert/status_effect/shower_regen/felinid
```

Nit

**Comment by Sakamoto4ka** [comment]
SS13 players don't know what a shower is

**Comment by necromanceranne** [comment]
Because it is felinid content rather than nothing at all, you definitely got permission, right?

**Comment by SomeoneYouProbablyKnow** [comment]
When is stam regen under a shower going to matter? If this is all just for the bit i respect it immensely

**Comment by itseasytosee** [comment]
This isn't really how our stamina system works. Stamina all returns in a lump sum 10 seconds after the last instance of stamina damage, with the timer being reset for each instance of stamina damage.

**Comment by Xackii** [comment]
> This isn't really how our stamina system works. Stamina all returns in a lump sum 10 seconds after the last instance of stamina damage, with the timer being reset for each instance of stamina damage. Taking a shower will not help you regenerate your stamina faster in game. The only real effect would be that if you get stamcrit under a shower you would be back on your feet in five seconds instead of 10, but you would still have heavy stamina damage until your time was up, so it would just make you easier to stunlock.
> This might actually have absurd combat implications specifically from the stamina damage from unarmed attacks, where it would be beneficial to drag the guy you are fighting under a shower so they can regenerate above the 100 stamina damage threshold so you can easily stamcrit them again, as the act of being stabcrit bumps your stamloss to 120. And i'm talking about normal humans here, not felinids.
> To be fair, this is more of an absurdity with the stamina system, but any effect that regens stam over time will play into that wackyness.

How do chemicals that restore stamina work in this case? If we perceive shower as some kind of reagent that restores each tick or takes away stamina, then this is a problem with stamina code, why does it work so strangely.

**Comment by itseasytosee** [comment]
Sorry, theres been a stamina refractor while I was gone that I didn't know about.
I don't know if this is going to work the way you intended but it wont do what i said it would.

**Comment by ZephyrTFA** [comment]
Given this has a species change and no maintainer has approved this I will be applying DNM until a maintainer steps forward who will sign off on this.

**Review by optimumtact — APPROVED** [review]

**Review by Wallemations — APPROVED** [review]
I look forward to seeing the halls completely covered in perpetually-running showers

**Pull request merged**
